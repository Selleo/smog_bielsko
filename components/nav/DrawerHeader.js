import React, { Component } from 'react';
import { StyleSheet, Text } from 'react-native';

export default class DrawerHeader extends Component {
  static get defaultProps() {
  }

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Text style={styles.text}>
        {this.props.text}
      </Text>
    )
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 18,
    color: '#222',
    paddingHorizontal: 20,
    paddingBottom: 20
  }
})
