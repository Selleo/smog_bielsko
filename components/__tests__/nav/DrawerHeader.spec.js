import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import DrawerHeader                 from '../../nav/DrawerHeader';

chai.use(dirtyChai);

let wrapper, props;

describe('DrawerHeader', () => {
  beforeEach(() => {
    props = {
      text: 'Wybierz miejscowosc'
    };

    wrapper = shallow(<DrawerHeader {...props}/>);
  });
  it('renders correctly', () => {
    expect(wrapper).to.exist();
  });

  it('render staton name on html', () => {
    expect(wrapper.children().text()).to.equal('Wybierz miejscowosc')
  });
});
