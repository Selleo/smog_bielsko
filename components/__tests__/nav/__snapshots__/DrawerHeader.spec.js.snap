exports[`DrawerHeader renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
        Object {
            "color": "#222",
            "fontSize": 18,
            "paddingBottom": 20,
            "paddingHorizontal": 20,
          }
    }>
    Wybierz miejscowosc
</Text>,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
            Object {
                  "color": "#222",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "paddingHorizontal": 20,
                }
      }>
      Wybierz miejscowosc
</Text>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DrawerHeader
        text="Wybierz miejscowosc" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DrawerHeader {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": "Wybierz miejscowosc",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
                    Object {
                              "color": "#222",
                              "fontSize": 18,
                              "paddingBottom": 20,
                              "paddingHorizontal": 20,
                            }
          }>
          Wybierz miejscowosc
</Text>,
        "_debugID": 2,
        "_renderedOutput": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
                    Object {
                              "color": "#222",
                              "fontSize": 18,
                              "paddingBottom": 20,
                              "paddingHorizontal": 20,
                            }
          }>
          Wybierz miejscowosc
</Text>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DrawerHeader
    text="Wybierz miejscowosc" />,
}
`;
